### PR TITLE
Add ability to search calculations by molecule name/inchi/etc

### PR DESCRIPTION
**Pull request by alesgenova**

### CODE REVIEW

File: girder/molecules/molecules/calculation.py [hunk 1]
```diff
@@ -391,61 +387,31 @@ def ingest_calc(self, calculation, body, detectBonds=None):
...
-        calcs = self._model.find(query, fields=fields, limit=limit,
-                                 offset=offset, sort=sort)
-        num_matches = calcs.collection.count_documents(query)
-
-        calcs = self._model.filterResultsByPermission(calcs, user,
-            AccessType.READ, limit=limit)
-        calcs = [self._model.filter(x, user) for x in calcs]
-
-        return search_results_dict(calcs, num_matches, limit, offset, sort)
+        return CalculationModel().findcal(moleculeId=moleculeId, imageName=imageName,
```
Comment (cjh1) [inline girder/molecules/molecules/calculation.py]: snake_case?

File: girder/molecules/molecules/models/calculation.py [hunk 2]
```diff
@@ -81,6 +87,76 @@ def validate(self, doc):
 
         return doc
 
+    def findcal(self, moleculeId=None, imageName=None,
```
Comment (cjh1) [inline girder/molecules/molecules/models/calculation.py]: case

File: girder/molecules/molecules/models/calculation.py [hunk 3]
```diff
@@ -81,6 +87,76 @@ def validate(self, doc):
...
+            if inchikey:
+                params['inchikey'] = inchikey
+            if smiles:
+                params['smiles'] = smiles
+            if formula:
+                params['formula'] = formula
+
+            molecules = MoleculeModel().findmol(params)['results']
+            moleculeIds = [molecule['_id'] for molecule in molecules]
+            query['moleculeId'] = {'$in': moleculeIds}
```
Comment (cjh1) [inline girder/molecules/molecules/models/calculation.py]: We can probably speed this up if necessary with an aggregation pipeline, when we get a large dataset imported we can see how it performs.

**Comment by psavery** [comment]
@cjh1 I fixed the `camel_case`. I'll need these changes also for doing some advanced searches with calculations.

**Pull request merged**